pennylane/circuit_graph.py

```diff
@@ -654,15 +542,14 @@ def update_node(self, old, new):
             raise ValueError("The new Operator must act on the same wires as the old one.")
         new.queue_idx = old.queue_idx
         nx.relabel_nodes(self._graph, {old: new}, copy=False)  # change the graph in place
+        self._operations = self.operations_in_order
+        self._observables = self.observables_in_order
 
-    def draw(
-        self, charset="unicode", show_variable_names=False, wire_order=None, show_all_wires=False
```

Comment:
Would we want to somehow keep the possibility to represent free parameters?

Reply:
The `show_variable_names`? This would be nice, but there are a couple of issues:

1. In the new core, we no longer perform any QNode signature inspection (this was one of the aims, as we were concerned about speed/efficiency/complicated logic around the signature inspection).
1. If you think about it, we are printing the _tape_, and *not* the QNode. We no longer make the assumption that QNode arguments == gate arguments, since classical processing is allowed within the QNode.


E.g.,

```python
@qml.qnode(dev)
def f(x):
    qml.RX(np.sin(x ** 3), wires=0)
    return qml.expval(qml.PauliZ(0))
```

How would we print this? we would not only need to inspect the signature of the qnode to get the string `x`, we would also need to somehow track the classical processing in order to print the gate argument as `np.sin(x ** 3)` 🤔 

This sounds like a lot of overhead/complicated logic for a feature that isn't often used.